src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -0,0 +1,345 @@
+                    # gRPC error is masked inside an artificial call,
+                    # caller will see this error if and only
+                    # if runs an `await call` operation
+                    return UnaryUnaryCallRpcError(err)
+                except asyncio.CancelledError:
+                    # Cancellation is masked inside an artificial call,
+                    # caller will see this error if and only
+                    # if runs an `await call` operation
+                    return UnaryUnaryCancelledError()
+
+                if isinstance(call_or_response, _base_call.UnaryUnaryCall):
+                    return call_or_response
+                else:
+                    return UnaryUnaryCallResponse(call_or_response)
+
+            else:
+                return UnaryUnaryCall(request,
+                                      _timeout_to_deadline(
+                                          self._loop,
+                                          client_call_details.timeout),
+                                      self._channel, client_call_details.method,
+                                      request_serializer, response_deserializer)
+
+        client_call_details = ClientCallDetails(method, timeout, None, None)
+        return await _run_interceptor(
+            iter(interceptors), client_call_details, request)
+
+    def cancel(self) -> bool:
+        if self._interceptors_task.done():
+            return False
+
+        self._interceptors_task.cancel()
+        return True
+
+    def cancelled(self) -> bool:
+        if not self._interceptors_task.done():
+            return False
+
+        call = self._interceptors_task.result()
+        return call.cancelled()
+
+    def done(self) -> bool:
+        if not self._interceptors_task.done():
+            return False
+
+        return True
+
+    def add_done_callback(self, unused_callback) -> None:
+        raise NotImplementedError()
+
+    def time_remaining(self) -> Optional[float]:
+        raise NotImplementedError()
+
+    async def initial_metadata(self) -> Optional[MetadataType]:
+        return await (await self._interceptors_task).initial_metadata()
+
+    async def trailing_metadata(self) -> Optional[MetadataType]:
+        return await (await self._interceptors_task).trailing_metadata()
+
+    async def code(self) -> grpc.StatusCode:
+        return await (await self._interceptors_task).code()
+
+    async def details(self) -> str:
+        return await (await self._interceptors_task).details()
+
+    async def debug_error_string(self) -> Optional[str]:
+        return await (await self._interceptors_task).debug_error_string()
+
+    def __await__(self):
+        call = yield from self._interceptors_task.__await__()
+        response = yield from call.__await__()
+        return response
+
+
+class UnaryUnaryCallRpcError(_base_call.UnaryUnaryCall):
+    """Final class that represents an UnaryUnaryCall that finished
+    with an RpcError."""
```

Comment:
nit: Can we make the first line of docstring doesn't wrap (<=80 characters)? This might be weird requirement, but the code base is following [Google Style Guide](http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).